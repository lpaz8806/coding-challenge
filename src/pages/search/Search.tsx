import {Box, Grid, Typography} from '@mui/material';
import { Observer } from 'mobx-react-lite';
import React, {ReactElement} from 'react';
import { withInstances } from '../../utils/withInstances';
import {ProductListView} from "./ProductListView";
import {SearchBoxView} from "./SearchBoxView";
import {ProductPreviewView} from "./ProductPreviewView";
import SearchViewModelType from "../../types/SearchViewModelType";

export function Search({ viewModel }:{ viewModel: SearchViewModelType }): ReactElement {
  return <Observer>{() =>
      <Box>
        <Grid container spacing={2}>
          <Grid item md={4}>
            <SearchBoxView
                onSearchTermChange={viewModel.setSearchValue}
                onSearchRequested={viewModel.search} />

            <ProductPreviewView product={viewModel.selectedProduct} />
          </Grid>

          <Grid item md={8}>
            <Box alignItems="center" flexGrow={1}>
              <Typography align="center" variant="h5">Search results</Typography>
            </Box>

            <ProductListView
                items={viewModel.searchResults}
                onProductSelected={viewModel.setSelectedProduct}
            />
          </Grid>
        </Grid>
      </Box>}
  </Observer>;
}

export default withInstances({ viewModel: 'SEARCH_PAGE' }, Search);
