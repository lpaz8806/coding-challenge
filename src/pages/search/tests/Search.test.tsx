import { render } from "@testing-library/react";
import Search from "../Search";

describe('SearchPage', () => {
  it('should render', () => {
    const { container } = render(<Search />);
    expect(container).toMatchSnapshot();
  });
});
