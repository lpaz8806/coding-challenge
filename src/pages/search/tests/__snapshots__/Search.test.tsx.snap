// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should render 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
      >
        <div
          class="css-m69qwo-MuiStack-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Search for an article
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-basic"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Search for an article
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1lddlii-MuiButtonBase-root-MuiButton-root"
            style="background-color: rgb(63, 81, 181); color: rgb(255, 255, 255);"
            tabindex="0"
            type="button"
          >
            Search
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-8 css-am41oj-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-1oaa9v5"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-ph2xsj-MuiTypography-root"
          >
            Search results
          </h5>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          No results
        </div>
      </div>
    </div>
  </div>
</div>
`;
